src/System.Private.ServiceModel/src/Internals/System/Runtime/IOThreadScheduler.cs

```diff
@@ -74,19 +74,22 @@ private IOThreadScheduler(int capacity)
 
             _slots = new Slot[capacity];
             Contract.Assert((_slots.Length & SlotMask) == 0, "Capacity must be a power of two.");
+
+            _overlapped = new ScheduledOverlapped();
         }
 
-        public static void ScheduleCallbackNoFlow(SendOrPostCallback callback, object state)
+        public static void ScheduleCallbackNoFlow(Action<object> callback, object state)
         {
             if (callback == null)
             {
-                throw Fx.Exception.ArgumentNull("callback");
+                throw Fx.Exception.ArgumentNull(nameof(callback));
             }
 
             bool queued = false;
             while (!queued)
             {
-                try { }
+                try
+                { }
                 finally
```

Comment:
Is this code run on platforms other than .NET Core? This try/finally pattern isn't applicable to core, which lacks thread aborts.